pkg/scheduler/queue/queue.go

```diff
@@ -399,15 +399,15 @@ type querierConnection struct {
 	element  *list.Element
```

Comment:
element is never read, do we need it at all

Reply:
Good catch, that was left over from a previous approach to removing requests when their context was cancelled. I've removed it.